quiche/src/recovery/bbr/per_transmit.rs

```diff
@@ -36,7 +36,9 @@ pub fn bbr_on_transmit(r: &mut Recovery) {
 
 // 4.3.4.4.  Restarting From Idle
 fn bbr_handle_restart_from_idle(r: &mut Recovery) {
-    if r.bytes_in_flight == 0 && r.delivery_rate.app_limited() {
```

Comment:
Interesting bug. So bbr_on_transmit() is called after bytes_in_flight is increased (https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/mod.rs#L293), there is no chance bytes_in_flight here is 0.

What about just moving https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/mod.rs#L293 after calling bbr_on_transmit() (L295)?

In linux tcp bbr, it is [checked](https://github.com/torvalds/linux/blob/master/net/ipv4/tcp_bbr.c#L336) when CA_EVENT_TX_START (first transmission when no packet is in flight), so I think it's good to keep the definition (checking bytes_in_flight == 0, not a small number) here.

Reply:
Thats actually much better. Thank you for reviewing the code change.